File: euler2.rb
# solution for http://projecteuler.net/problem=2

i = 0
fibo = Array.new
fibo = [1, 1, 2, 3, 5, 8]
lodo = Array.new

# -1 would be the last index position of the Array

# first step. generate the fibonacci numbers
while (fibo[-1] < 4000000)
	p = fibo[-1] + fibo[-2]
	puts p # debug thing
	fibo.push p
end	

fibo.each_with_index do |e, i|
	if e % 2 == 0
		m = e
		lodo.push m
	else
		puts "odd number"
end	
puts "the numbers in lodo are:" 
puts lodo

puts "...calculating"

sum = lodo.reduce :+
puts sum	
end

File: euler2.swift
# http://projecteuler.net/problem=2

var fibonacciSequence: Int[] = [1,2,3,5,8]
var fiboIndex = fibonacciSequence.count-1
while fibonacciSequence[fiboIndex] < 4_000_000 {
    fibonacciSequence.append(fibonacciSequence[fiboIndex-1] + fibonacciSequence[fiboIndex])
    fiboIndex = fibonacciSequence.count-1
}

// println("\(fibonacciSequence[fiboIndex-1])")

var accumulator = 0
for number in fibonacciSequence {
    if number % 2 == 0 {
        accumulator += number
    } else {
        accumulator += 0
    }
}
println(accumulator)